### DISCUSSION TITLE
How to get Touch working with ?Resistive Touch?  Not sure

### DISCUSSION THREAD

**Discussion by dookie454** [Bodmer/TFT_eSPI#1020] [Q&A - Touch] [answered]
Hello!  This library is great,  very FAST, which is what I was looking for!  So thanks for that!

Now, trying to get Touch working, and cant.  I get no change in the X and y values, they show 0.  

I'm using only the 1 wire (T_CS) for touch, wired to GPIO 21.  

This display:  https://www.amazon.com/gp/product/B073R7BH1B/ref=ppx_yo_dt_b_asin_title_o00_s00?ie=UTF8&psc=1
HiLetgo ILI9341 2.8" SPI TFT LCD Display Touch Panel 240X320 with PCB 5V/3.3V STM32

I have ESP32 Devkit v1, which doesnt have the GPIO 0 port like the v4 has.  No biggie, incase you were wondring why I didnt use 0.

// For ESP32 Dev board (only tested with ILI9341 display)
// The hardware SPI can be mapped to any pins
#define TFT_MISO 19  //SDD on LCD
#define TFT_MOSI 23  //SDI on LCD
#define TFT_SCLK 18  //SCK on LCD
#define TFT_CS   15  //CS on LCD - Chip select control pin
#define TFT_DC    2  //DC on LCD - Data Command control pin
#define TFT_RST  27  //0 works for Devkit v4, was 4  RESET pin on LCD -  Reset pin (could connect to RST pin).  ESP32 Devkitv1 does not have pin 0!
#define TOUCH_CS 21     // Chip select pin (T_CS) of touch screen  - THIS DOES NOT WORK.

Thanks for any help you have for me,  everything else is working GREAT!!

**Comment by Bodmer** [comment, marked as answer]
Have you connected the touch controller pins to the SPI bus?

T_DO -> TFT_MISO pin
T_DIN -> TFT_MOSI pin
T_CLK -> TFT_SCLK pin

Run the Test_Touch_Controller diagnostic example to test the connections.

**Reply by dookie454** [reply]
Did not connect those,  probably because I only saw something about adding the Touch_CS pin to the ESP32 : )

It works now!

Thanks!!
